models/balancer/arbitrum/balancer_v2_arbitrum_trades.sql

```diff
@@ -0,0 +1,228 @@
+    )
+}}
+
+{% set project_start_date = '2021-08-27' %}
+
+WITH    
+    fees_base AS (SELECT *, MAX(index) OVER(PARTITION BY tx_hash, contract_address) AS max_index_same_tx FROM balancer_v2_arbitrum.pools_fees),
+    most_case_fees AS (SELECT * FROM fees_base WHERE index = max_index_same_tx),
+    edge_case_fees AS (
+        SELECT 
+            s.evt_block_number,
+            s.evt_tx_hash,
+            s.evt_index,
+            f.swap_fee_percentage 
+        FROM balancer_v2_arbitrum.Vault_evt_Swap s
+        INNER JOIN fees_base f 
+        ON s.evt_tx_hash = f.tx_hash AND f.index < s.evt_index
+    ),
+    swap_fees AS (
+        SELECT
+            swap.evt_block_number,
+            swap.evt_tx_hash,
+            swap.evt_index,
+            SUBSTRING(swap.poolId, 0, 42) AS contract_address,
+            swap.evt_block_time,
+            MAX(fees.block_time) AS max_fee_evt_block_time
+        FROM
+            {{ source ('balancer_v2_arbitrum', 'Vault_evt_Swap') }} swap
+            LEFT JOIN {{ ref('balancer_v2_arbitrum_pools_fees') }} fees
+                ON fees.contract_address = SUBSTRING(swap.poolId, 0, 42)
+                AND fees.block_time <= swap.evt_block_time
+        GROUP BY 1, 2, 3, 4, 5
+    ),
+    dexs AS (
+        SELECT
+            swap.evt_block_number,
+            swap.evt_block_time AS block_time,
+            '' AS taker,
+            '' AS maker,
+            swap.amountOut AS token_bought_amount_raw,
+            swap.amountIn AS token_sold_amount_raw,
+            CAST(NULL as DOUBLE) AS amount_usd,
+            swap.tokenOut AS token_bought_address,
+            swap.tokenIn AS token_sold_address,
+            swap_fees.contract_address AS project_contract_address,
+            swap.poolId AS poolId,
+            COALESCE(most.swap_fee_percentage, edge.swap_fee_percentage) / POWER(10, 18) AS swap_fee,
+            swap.evt_tx_hash AS tx_hash,
+            '' AS trace_address,
+            swap.evt_index
+        FROM
+            swap_fees
+            INNER JOIN {{ source ('balancer_v2_arbitrum', 'Vault_evt_Swap') }} swap
+                ON swap.evt_block_number = swap_fees.evt_block_number
+                AND swap.evt_tx_hash = swap_fees.evt_tx_hash
+                AND swap.evt_index = swap_fees.evt_index
+            LEFT JOIN most_case_fees most
+                ON most.contract_address = swap_fees.contract_address
+                AND most.block_time = swap_fees.max_fee_evt_block_time
+                AND most.tx_hash <> swap_fees.evt_tx_hash
+            LEFT JOIN edge_case_fees edge
+                ON edge.evt_block_number = swap_fees.evt_block_number
+                AND edge.evt_tx_hash = swap_fees.evt_tx_hash
+                AND edge.evt_index = swap_fees.evt_index
+        WHERE
+            swap.tokenIn <> swap_fees.contract_address
+            AND swap.tokenOut <> swap_fees.contract_address
+            {% if is_incremental() %}
```

Comment:
can the incremental filters be put in the CTE called?

Reply:
Yes, done.